decidim-core/lib/tasks/decidim_robots.rake

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :robots do
+    desc "Overrides robots.txt with a custom one."
+    task :replace, [] => :environment do
+      actions :append_file, "public/robots.txt", <<~SQUISH
```

Comment:
As we're describing as an override, I'd prefer to just override the `robots.txt` file. That would allow us to reuse this same task for other changes in this file in the future.

Reply:
TBH, i would dedicate some time to rewrite the app generator, as currently we are generating a rails app, then apply a lot of patches to the existing config. like the application.rb patch and so on ... like i did here... https://github.com/tremend-cofe/decidim/pull/278

Reply:
By the way, this is a Rake task that can work for upgrade. If we follow the generator path, no old application would be able to have this update, unless manually intervened .

Reply:
is not very clear for me what you expect here. IMO, we could change the description of the task to something like "Append Decidim rules to robotx.txt". 

But i would still do it as a template part of the generator.

Reply:
What I mean is to change the the `append_file`, I've left it on https://github.com/decidim/decidim/pull/11693#discussion_r1360276600 

> TBH, i would dedicate some time to rewrite the app generator, as currently we are generating a rails app, then apply a lot of patches to the existing config. 

Ok, but I don't think that'd be relevant here, as the initial idea of what I had in mind is to be able to `backport` this one. Seeing the final result I'm not 100% sure this is a `type: fix`, but how do you see it?

Reply:
Let's keep it as a change.